weaviate/connect/base.py

```diff
@@ -118,16 +118,24 @@ def _grpc_channel(self, proxies: Dict[str, str], grpc_msg_size: Optional[int]) -
             options: list = [*opts, ("grpc.http_proxy", p)]
         else:
             options = opts
+
+        # Add environment-based logging interceptor
+        from weaviate.logger import GrpcLoggingInterceptor
+
+        interceptors = [GrpcLoggingInterceptor()]
+
         if self.grpc.secure:
             return grpc.aio.secure_channel(
                 target=self._grpc_target,
                 credentials=ssl_channel_credentials(),
                 options=options,
+                interceptors=cast(Sequence[Any], interceptors),
```

Comment:
To simplify this, I'd recommend annotating line 125 like so:
```python
interceptors: Sequence[Any] = [GrpcLoggingInterceptor()]
```
so that there is no need for two `cast` calls here and on line 138